### PR TITLE
Clean up ESLint configuration

### LINKED ISSUE #20: Move ESLint configuration from command arguments to .eslintrc.json
#### Description
Currently, our `yarn lint` command calls `eslint` with a [series of configuration options as inline arguments](https://github.com/tvkitchen/base/blob/d7e3c1d4da5470b9d4a07a2cb4ff3b385ebbb9af/package.json#L11).

These should all be moved to [the ESLint configuration file](https://github.com/tvkitchen/base/blob/d7e3c1d4da5470b9d4a07a2cb4ff3b385ebbb9af/.eslintrc), which we should also rename to `.eslintrc.json` (as the format-less version is [deprecated](https://eslint.org/docs/user-guide/configuring#configuration-file-formats)).

### PR DESCRIPTION
**Pull request by reefdog**
## Description

This minor PR:

- Renames our ESLint configuration file from the deprecated format-less `.eslintrc` to the more explicit `.eslintrc.json`
- Moves our ESLint configuration from inline invocation arguments in `package.json` to the `.eslintrc.json` configuration file

## Due Diligence Checklist
- [x] Tests
- [x] Documentation
- [x] Consolidated commits
- [x] Relevant labels assigned

## Steps to Test
1. `yarn lint`

## Related Issues
Resolves #20

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -11,5 +11,6 @@
     "es6": true,
     "node": true,
     "jest": true
-  }
+  },
+  "ignorePatterns": ["lib/"]
```
Comment (slifty) [inline .eslintrc.json:15]: should we also ignore `node_modules` or is it smart enough to do that on its own.

Comment (reefdog) [reply .eslintrc.json:15]: o rite I buried that in the [commit description](https://github.com/tvkitchen/base/pull/27/commits/a1f9e3bfb9b0ff9e8e5367cea0d50414ed51eb65). Yes, it does!

Comment (slifty) [reply .eslintrc.json:15]: I'm supposed to read COMMENT DESCRIPTIONS now?!?!?!!??!?!1

Comment (reefdog) [reply .eslintrc.json:15]: Nope! My fault for not elevating that to PR description. (For real I don't think it's reasonable to assume you dug into each commit.)

Comment (slifty) [reply .eslintrc.json:15]: So now I'm supposed to read PR DESCRIPTIONS?!?!?!?!?!?!?!1

Comment (reefdog) [reply .eslintrc.json:15]: Well, less "now" than "then and also always"…

**Pull request merged**